src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -629,13 +631,11 @@ private void SetDefaultProjectName()
 
         private async Task EnsureNuGetAndVsProjectAdapterCacheAsync()
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
almost all the method needs to be on UI thread or most of it's called APIs also needs to be on UI thread so I'm not sure how much performance we gained moving it off the UI thread?